shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java

```diff
@@ -0,0 +1,95 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.android;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.widget.FrameLayout;
+
+/**
+ * Displays a Flutter UI on an Android device.
+ * <p>
+ * A {@code FlutterView}'s UI is painted by a corresponding {@link FlutterEngine}.
+ * <p>
+ * A {@code FlutterView} can operate in 2 different {@link RenderMode}s:
+ * <ol>
+ *   <li>{@link RenderMode#surface}, which paints a Flutter UI to a {@link android.view.SurfaceView}.
+ *   This mode has the best performance, but a {@code FlutterView} in this mode cannot be positioned
+ *   between 2 other Android {@code View}s in the z-index, nor can it be animated/transformed.</li>
+ *   <li>{@link RenderMode#texture}, which paints a Flutter UI to a {@link android.graphics.SurfaceTexture}.
+ *   This mode is not as performant as {@link RenderMode#surface}, but a {@code FlutterView} in this
+ *   mode can be animated and transformed, as well as positioned in the z-index between 2+ other
+ *   Android {@code Views}.</li>
```

Comment:
nit: I haven't run any actual benchmarks on this, but my understanding is that `SurfaceView`s are so much more performant than `TextureView`s that it probably makes sense for us to recommend them as a default unless they really need the Flutter UI to fit into the Android UI view hierarchy better. WDYT?

I think it's also probably worth linking to the official Android docs here for some more in depth reading if people are interested.

https://source.android.com/devices/graphics/arch-tv#surface_or_texture

Reply:
Comments updated.